src/SourceLocator/SourceStubber.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+namespace BetterReflection\SourceLocator;
+
+use PhpParser\Node\Stmt\Class_;
+use PhpParser\Node\Stmt\Interface_;
+use PhpParser\Node\Stmt\Trait_;
+use PhpParser\ParserFactory;
+use PhpParser\PrettyPrinter\Standard;
+use Zend\Code\Generator\ClassGenerator;
+use Zend\Code\Reflection\ClassReflection;
+
+/**
+ * Function that generates a stub source from a given reflection instance.
+ */
+final class SourceStubber
+{
+    /**
+     * @var \PhpParser\Parser
+     */
+    private $parser;
+
+    /**
+     * @var Standard
+     */
+    private $prettyPrinter;
+
+    public function __construct()
+    {
+        $this->parser        = (new ParserFactory())->create(ParserFactory::PREFER_PHP7);
+        $this->prettyPrinter = new Standard();
+    }
+
+    /**
+     * @param ClassReflection $reflection
+     *
+     * @return string
+     */
+    public function __invoke(ClassReflection $reflection)
+    {
+        $stubCode   = ClassGenerator::fromReflection($reflection)->generate();
+        $interface  = $reflection->isInterface();
```

Comment:
Naming: imo, booleans should be `$isInterface` as communicates to the reader this should be `bool`